src/Screens/Main/Settings/Password.tsx

```diff
@@ -46,21 +46,34 @@ export default ({ navigation }: Props) => {
     repeatedNewPassword.length > 0 &&
     newPassword === repeatedNewPassword;
 
+  const onButtonPress = () => {
+    dispatch(state.changePassword({ currentPassword, newPassword }));
+  };
+
   const onGoBack = () => {
     navigation.goBack();
   };
 
-  const onButtonPress = () => {
-    dispatch(state.changePassword({ currentPassword, newPassword }));
+  const resetAndGoBack = () => {
+    setCurrentPassword('');
+    setNewPassword('');
+    setRepeatedNewPassword('');
+    dispatch(state.reset);
+    onGoBack();
   };
+
   const requestState = useSelector(state.select);
 
+  const changeIsResolved =
+    RD.isSuccess(requestState) || RD.isFailure(requestState);
+
   React.useEffect(() => {
-    if (RD.isSuccess(requestState)) {
-      setCurrentPassword('');
-      setNewPassword('');
-      setRepeatedNewPassword('');
-      const timeout = setTimeout(onGoBack, state.coolDownDuration);
+    if (changeIsResolved) {
+      const callBack = RD.isSuccess(requestState)
+        ? resetAndGoBack
+        : dispatch(state.reset);
```

Comment:
In this spot there is way too much complexity.

Reply:
Switch case success/ Failure?

Split to two effects?

Is it good ux to handle failed req automatically. Its kinda rare case. It sucks to reset that state. Imo this is bad way to handle it.

Better and simpler way is just to give the user an option to retry on failure.

Reply:
![Screenshot from 2021-06-27 16-27-38](https://user-images.githubusercontent.com/34128180/123546705-653c0380-d766-11eb-92d3-d624dc723557.png)

Great. I simplified this a bit. Now only resetting the requestState if it is successfull. It will show the success-message and then fade away. In case of failure, this failure-dialog will show. User can try again with the try-again button then.

Reply:
@pheis pls check this PR now 8)